src/mixins/socket.js

```diff
@@ -59,7 +59,17 @@ export default {
             this.$router.push("/setup")
         });
 
-        socket.on('monitorList', (data) => {
+        socket.on("monitorList", (data) => {
+            // Add Helper function
+            Object.entries(data).forEach(([monitorID, monitor]) => {
+                monitor.getUrl = () => {
+                    try {
+                        return new URL(monitor.url);
+                    } catch (_) {
+                        return null;
+                    }
```

Comment:
Why this code is duplicated with `getUrl` in `monitor.js`?

Reply:
Since one is used in frontend and one is used in backend, even if I extract it to `util.js` i'll have to duplicate like `ucfirst()` did. Any thoughts?

Reply:
I actually want to ask too, are there any way that I can share both function in frontend and backend. Since frontend is using ES6 Import syntax, and backend (node.js) is using "require/export"

Reply:
Referring to [this](https://stackoverflow.com/a/45854500), we can add `{ "type": "module" }` to `package.json` and convert all commonjs requires to ES6 imports.

Reply:
> Referring to [this](https://stackoverflow.com/a/45854500), we can add `{ "type": "module" }` to `package.json` and convert all commonjs requires to ES6 imports.

As I remember, I did try before, some nodejs packages cannot be imported without "require()"

Reply:
Maybe we can use TypeScript for sharing code between frontend and backend, but need some time to implement. It is OK to duplicate a common function for now.

Reply:
Switching to Typescript would honestly be my preference too.